src/zimscraperlib/video/encoding.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# vim: ai ts=4 sts=4 et sw=4 nu
+
+
+import subprocess
+from . import logger
+from ..logging import nicer_args_join
+
+
+def reencode(src_path, dst_path, config, delete_src=False, return_output=False):
+    """Runs ffmpeg with given config and returns back the returncode received from ffmpeg
+    
+    config - A list of ffmpeg arguments (except -threads and -max_muxing_queue_size)"""
+
+    tmp_path = src_path.parent.joinpath(f"video.tmp{dst_path.suffix}")
+    args = ["ffmpeg", "-y", "-i", f"file:{src_path}"] + config + [f"file:{tmp_path}"]
+    logger.info(f"Encode {src_path} -> {dst_path} video format = {dst_path.suffix}")
+    logger.debug(nicer_args_join(args))
+    ffmpeg = subprocess.run(args, stderr=subprocess.PIPE, universal_newlines=True)
+    if ffmpeg.returncode == 0:
+        if delete_src:
+            src_path.unlink()
+        tmp_path.replace(dst_path)
+    else:
+        if tmp_path.exists():
+            tmp_path.unlink()
+    if return_output:
```

Comment:
Thanks for this. Works as expected.
One advice: when returning different number of args, always return in same order.
Here, you should return the returncode first (so it's the first in any case), then the output (only in certain case).

But different number of args is not a very good pattern anyway. Keeping the goal in mind (knowing if it worked, having the ability to look at output), I'd like to propose the following:

```py
def reencode(src_path, dst_path, config, delete_src=False, failsafe=True):
    # begining stays the same

    if not failsafe:
        ffmpeg.check_returncode()

    return ffmpeg.returncode == 0, ffmpeg

# usage
reencode(src, dst, args)
success, _ = reencode(src, dst, args)
success, process = reencode(src, dst, args)
print(process.stdout)
try:
    reencode(src, dst, args, failsafe=False)
except subprocess.CalledProcessError as exc:
    print("Error", exc.returncode)
    print(exc.stdout)
```

what do you think? We could still have a `with_process=False` param that if set returns this tuple and if not just the success part… If 99% of use cases is `if reencode(src, dst, args):` that might make more sense.

Reply:
It's actually a cool way to handle it IMO. I've added the with_process option too. Thanks.